frontend/app/src/app/_services/account.service.ts

```diff
@@ -1,17 +1,65 @@
 import { Injectable } from '@angular/core';
-//import { HttpClient } from '@angular/common/http';
+import { HttpClient } from '@angular/common/http';
 import { BehaviorSubject, Observable, of  } from 'rxjs';
+import { Router } from '@angular/router';
+import { environment } from '../../environments/environment';
+import { map } from 'rxjs/operators';
+
 import { User } from '../_models/user';
 import { USER } from '../_mock_models/mock_user';
 
 @Injectable({
   providedIn: 'root'
 })
 export class AccountService {
+  user : User;
+  constructor( private router: Router) {
+   }
+
+  getUser(): User {
+    return this.user;
+  }
+
+  setUser(){
+    this.user = USER;
+  }
+
+  logout(){
+    this.user = null;
+    this.router.navigate(['/login']);
+  }
+
+  /*
+  private userSubject: BehaviorSubject<User>;
+  public user: Observable<User>;
+
+  constructor(
+    private router: Router,
+    private http: HttpClient
+  ) {
+      this.userSubject = new BehaviorSubject<User>(JSON.parse(localStorage.getItem('user')));
+      this.user = this.userSubject.asObservable();
+  }
 
-  constructor() { }
+  public get userValue(): User {
+    return this.userSubject.value;
+  }
 
-  getUser(): Observable<User> {
-    return of(USER);
+  login(){
+    return this.http.post<User>(`${environment.apiUrl}/users/authenticate`, { username })
+    .pipe(map(user => {
+        // store user details and jwt token in local storage to keep user logged in between page refreshes
+        localStorage.setItem('user', JSON.stringify(user));
```

Comment:
As a general rule, security/session tokens should go into https://developer.mozilla.org/en-US/docs/Web/API/Window/sessionStorage i.e. sessionStorage instead of localStorage.